api/src/main/java/org/apache/cloudstack/api/command/user/template/GetUploadParamsForTemplateCmd.java

```diff
@@ -168,6 +170,12 @@ private void validateRequest() {
         if (getZoneId() <= 0) {
             throw new ServerApiException(ApiErrorCode.PARAM_ERROR, "invalid zoneid");
         }
+        if (!hypervisor.equalsIgnoreCase(Hypervisor.HypervisorType.VMware.toString()) && osTypeId == null) {
+            throw new ServerApiException(ApiErrorCode.PARAM_ERROR, "Missing parameter ostypeid");
+        }
+        if (hypervisor.equalsIgnoreCase(Hypervisor.HypervisorType.VMware.toString()) && osTypeId != null) {
```

Comment:
Should we simply ignore for VMware, even if ostypeid is provided?

Reply:
... read the code down below, looks like we want the ostypeid to be null so we can get  it from `getDefaultDeployAsIsGuestOsId`

Reply:
should we really refuse the parameter as opposed to just ignore it? being this strict may crash operator scripts unnecessarily.

Reply:
guess I'm late too the party. see ^^

Reply:
This change is in line with the changes we have done with the registerTemplate API. We are informing the user in case they provide the 'ostypeid' parameter that it is not used anymore and failing